### PR TITLE
squeeze & unsqueeze operators

### PR DESCRIPTION
**Pull request by StijnWoestenborghs**
- Adding the squeeze operator
- Adding the unsqueeze operator
- Unittesting both

NOTE: Merging the old squeeze and unsqueeze branches here, as they had conflicting changes on Attributes

### CODE REVIEW

**Comment by StijnWoestenborghs** [comment]
I think calculating the result shape of squeeze and unsqueeze can be simplified a lot, let me try

**Comment by StijnWoestenborghs** [comment]
- Its maybe a little weird that specifying dim is stored as TensorShape in Attribute. 
- Supporting multiple dims in unsqueeze is a litlle more complicated as it would most likely assume a sorted list to work. Also interesting that [torch.unsqueeze](https://pytorch.org/docs/stable/generated/torch.unsqueeze.html#torch.unsqueeze) support only 1 value here and [jax.expand_dims](https://jax.readthedocs.io/en/latest/_autosummary/jax.numpy.expand_dims.html) does support a tuple

**Comment by andresnowak** [comment]
The only thing is that when one only passes one dimension (without using tensors shape or static_tuple), the way the value is saved in attributes is saved is different and the value can't be converted to tensorshape, at least not easily.

**Comment by StijnWoestenborghs** [comment]
I think changing attributes is probably the right call here. If we do that, the API of attributes you just take in and return a Tuple/List or something. (also @Benny-Nottonson also wanted to get rid of `data_shape` in attributes completely.)

**Comment by StijnWoestenborghs** [comment]
I think a hacky way now could be, changing attributes to:
```
    fn to_shape(self) -> TensorShape:
        if self.data_shape == StaticIntTuple[MAX_RANK]():
            return TensorShape(self.data[0].to_int())
        else:
            return TensorShape(rank=self.data[0].to_int(), shape=self.data_shape)
```
But I think its better to remove `data_shape` and `to_shape() -> TensorShape` completely from Attributes and have from list and `to_list` method that return a list of one if only one scalar/int was specified.

**Comment by StijnWoestenborghs** [comment]
Alright for me this is OK actually. Having attributes store a list could be a separate PR imo.

**Review by Benny-Nottonson — APPROVED** [review]
I agree, looks good to me

**Comment by andresnowak** [comment]
I just wanted to add a last test for unsqueeze with torch, but because unsqueeze behaves differently i don't know what other alternative to use, one is reshape but i don't know if to use another one.

**Comment by andresnowak** [comment]
Okay now i think it can be merged

**Review by andresnowak — APPROVED** [review]
